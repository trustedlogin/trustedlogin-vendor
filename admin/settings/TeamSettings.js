import { useMemo, useState } from "react";
import { __ } from "@wordpress/i18n";
import { FormTable, Input, Select } from "./components";
import RoleMultiSelect from "./RoleMultiSelect";

/**
 * Settings for one single team
 */
const TeamSettings = ({ team, setTeam }) => {
  const teamId = useMemo(() => {
    return team.id;
  }, [team]);


  const helpDeskOptions = [
    {
      label: __("Helpscout"),
      value: "helpscout",
    },
  ];
  return (
    <FormTable title={__("Team")}>
      <Input
        label={__("TrustedLogin Account ID")}
        name={`team-${teamId}[account_id]`}
        value={team.account_id}
        onChange={(value) =>
          setTeam({
            ...team,
            account_id: value,
          })
        }
      />
      <Input
        label={__("TrustedLogin API Key")}
        name={`team-${teamId}[api_key]`}
        value={team.api_key}
        onChange={(value) =>
          setTeam({
            ...team,
            api_key: value,
          })
        }
      />
      <Input
        label={__("TrustedLogin Private Key")}
        name={`team-${teamId}[private_key]`}
        value={team.private_key}
        onChange={(value) =>
          setTeam({
            ...team,
            private_key: value,
          })
        }
      />
      <RoleMultiSelect
        label={__("What user roles provide support?")}
        help={__(
          "Which users should be able to log into customersâ€™ sites if they have an Access Key?"
        )}
        name={`team-${teamId}[approved_roles]`}
        approvedRoles={team.approved_roles}
        onChange={(value) =>
          setTeam({
            ...team,
            approved_roles: value,
          })
        }
      />
      <Select
        label={__("Which helpdesk software are you using?")}
        name={`team-${teamId}[helpdesk]`}
        value={team.helpdesk}
        options={helpDeskOptions}
        onChange={(value) =>
          setTeam({
            ...team,
            helpdesk: value,
          })
        }
      />
    </FormTable>
  );
};

export default TeamSettings;
