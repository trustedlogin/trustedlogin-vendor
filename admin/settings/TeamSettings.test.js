import { render, fireEvent, cleanup, act } from "@testing-library/react";
import TeamSettings from "./TeamSettings";

describe("TeamSettings", () => {
  afterEach(cleanup);
  let team = {
    id: 0,
    account_id: "7",
    private_key: "pk_1",
    api_key: "ak_1",
    helpdesk: "helpscout",
    approved_roles: ["editor"],
  };
  it("renders", () => {
    const { container } = render(<TeamSettings team={team} />);
    expect(container).toMatchSnapshot();
  });

  it("updates", () => {
    const setTeam = jest.fn();
    const { container, getByLabelText } = render(
      <TeamSettings team={team} setTeam={setTeam} />
    );
    //Change an input
    act(() => {
      fireEvent.change(getByLabelText("TrustedLogin Account ID"), {
        target: { value: "42" },
      });
    });

    expect(setTeam).toBeCalledWith({
      ...team,
      account_id: "42",
    });
  });
});
