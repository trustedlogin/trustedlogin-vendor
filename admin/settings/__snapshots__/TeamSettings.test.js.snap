// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSettings renders 1`] = `
<div>
  <h2
    class="title"
  >
    Team
  </h2>
  <table
    class="form-table"
    role="presentation"
  >
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <label
            for="team-0[account_id]"
          >
            TrustedLogin Account ID
          </label>
        </th>
        <td>
          <input
            class="regular-text ltr"
            id="team-0[account_id]"
            name="team-0[account_id]"
            type="input"
            value="7"
          />
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <label
            for="team-0[api_key]"
          >
            TrustedLogin API Key
          </label>
        </th>
        <td>
          <input
            class="regular-text ltr"
            id="team-0[api_key]"
            name="team-0[api_key]"
            type="input"
            value="ak_1"
          />
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <label
            for="team-0[private_key]"
          >
            TrustedLogin Private Key
          </label>
        </th>
        <td>
          <input
            class="regular-text ltr"
            id="team-0[private_key]"
            name="team-0[private_key]"
            type="input"
            value="pk_1"
          />
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <label
            for="team-0[approved_roles]"
          >
            What user roles provide support?
          </label>
          <span
            class="description"
            id="team-0[approved_roles]-description"
          >
            Which users should be able to log into customersâ€™ sites if they have an Access Key?
          </span>
        </th>
        <td
          data-children-count="1"
        >
          <select />
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <label
            for="team-0[helpdesk]"
          >
            Which helpdesk software are you using?
          </label>
        </th>
        <td
          data-children-count="1"
        >
          <select>
            <option
              class="level-0"
              value="helpscout"
            >
              Helpscout
            </option>
          </select>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
